### ISSUE TITLE
Data appears incomplete when compared to other DCC fed sources.

### ISSUE THREAD

**Issue by adyoull** [HandyHat/ha-hildebrandglow-dcc#305]
**Describe the bug**
Read out is always less than what actually using from iHD, and Loop Energy DCC driven system.

**Comment by smithbill17** [comment]
I think this is the same as issue #292 where the accumulating cost appears to go down/reset at approximately 1.30am, resulting in the days total cost being lower than it otherwise should be.

**Comment by adyoull** [comment]
> I think this is the same as issue #292 where the accumulating cost appears to go down/reset at approximately 1.30am, resulting in the days total cost being lower than it otherwise should be.

Homes assistant is reporting 18.1kWh usage yesterday Loop energy and iHD are reporting 26.3kWh that's a significant difference, and my EV charger is reporting 17kWh usage yesterday just to charge my car.

**Comment by smithbill17** [comment]
Do you charge your car overnight? If HomeAssistant is losing/resetting it's energy accumulation around 1.30am, then perhaps that would account for the significant discrepancy lost while charging your car.  Presumably if you don't charge your car (or avoid using energy 12am-2am) then the discrepancy would be much smaller.

I think the developer is investigating (hopefully).

Is the integration losing/resetting the kWh consumed? Or is it just the cost of the kWh consumed?

**Comment by smithbill17** [comment]
Looking at my own history charts & those posted by others in issue #292 , it seems to be the kWh consumed anywhere between 12am & around 8am (or possibly all the way up to 12pm) are lost/reset.  So if you are charging your car (or other significant consumption) overnight, then that would be why you see a significant discrepancy.

Hopefully @HandyHat is on the case.

**Comment by ColinRobbins** [comment]
I am seeing this too.  The new algorithm is considerably under reporting compared with the bright app.
I think the issue may be the new algorithm does not “catch up” if there is a missed reading.  The original algorithm caught up so was accurate, albeit putting the consumption in the wrong time slot.

**Comment by Davrosx** [comment]
Whether this issue is linked to the resetting of data of not, my DCC data seems to begin the day increasing normally until between 0102-0104 hours and then stops with various issues (readings going up/down and remaining flat) until resuming at a later point in the day when it fails to catch up with the DCC recorded totals. 

I am also using a template sensor to sum the majority of my home's usage through kasa and tapo wifi plugs combined with powercalc integration to do my lights and appliances with stable/constant usage. When I map both and compare with bright app readings both come up short but the DCC via HA is dreadful.  Bright app for the 6th is 8.98, 7th 6.804, 8th 6.735
The system has gone until 1733 hours before starting to increase normally but never catch up.

![image](https://user-images.githubusercontent.com/75336029/217821857-f73a49df-19a3-49e6-a196-c8a5e3a14738.png)

**Comment by Markbez** [comment]
> Looking at my own history charts & those posted by others in issue #292 , it seems to be the kWh consumed anywhere between 12am & around 8am (or possibly all the way up to 12pm) are lost/reset. So if you are charging your car (or other significant consumption) overnight, then that would be why you see a significant discrepancy.
> 
> Hopefully @HandyHat is on the case.

I have similar issues. Big gaps in the data and the daily totals are way out compared to actual usage. 

This was yesterday, bearing in mind we have storage heaters and most of our energy use is between 00.30 and 07.30!

![image](https://user-images.githubusercontent.com/47301398/217851541-429a5d5e-c31f-4df0-b106-ca6e632dcb0b.png)

**Comment by Mortalitis** [comment]
Also seeing large gaps in my data. Last time I had a full day of data was 23rd Jan. Mostly seems to be missing between 2am to around 8am ±1hr. Today is particularly bad as I only have 4 data points so far (00:00, 01:00, 12:00 and 18:00).

**Comment by zbb67** [comment]
I too have gaps in the data, just as @Markbez graph above and others.
In my case at least, during the day I also have gaps in the data on the phone app as well, so I think for me it is possibly a server side issue (I emailed Hildebrand support yesterday to ask if it was a widespread issue, awaiting their reply - although I appreciate API access to their servers might not be a priority for them).  
Interestingly, sometime later the missing data in the app seems to get backfilled, but this doesn't also backfill into Home Assistant.

**Comment by zbb67** [comment]
> I emailed Hildebrand support yesterday to ask if it was a widespread issue

I had a detailed and friendly reply, it seems that the data gaps are a capacity issue due to the popularity of the service, they are upgrading the infrastructure as soon as they can.

**Comment by changeofaspect** [comment]
> > Looking at my own history charts & those posted by others in issue #292 , it seems to be the kWh consumed anywhere between 12am & around 8am (or possibly all the way up to 12pm) are lost/reset. So if you are charging your car (or other significant consumption) overnight, then that would be why you see a significant discrepancy.
> > Hopefully @HandyHat is on the case.
> 
> I have similar issues. Big gaps in the data and the daily totals are way out compared to actual usage.
> 
> This was yesterday, bearing in mind we have storage heaters and most of our energy use is between 00.30 and 07.30!
> 
> ![image](https://user-images.githubusercontent.com/47301398/217851541-429a5d5e-c31f-4df0-b106-ca6e632dcb0b.png)

My graphs look just like yours and my HA is only showing about 15~20% of my genuine consumption. This situation is getting worse every day and yesterday I kept watching the Bright App to see what happened. At 09:00, the App showed the first two half hour periods. It did not change state until 18:30 when it started adding in the data for the early hours. HA mirrored this with a flatline from 01:00 to 18:00 and then slowly increasing each half hour to midnight. At midnight, the HA integration did it's reset but the Bright App was only showing data to 08:00, around 16hours of lag. As I write this, the Bright App is showing the first two half hours of today, but yesterday is incomplete with data from midnight to 11:30. I am still waiting for 12hours worth of data from yesterday. I know the data is there in the DCC as my suppliers website shows it.

**Comment by Carl-mgb66** [comment]
Similar issues here. It's making the integration very unhelpful. Note that both the Bright app and my supplier are showing the right data for yesterday now, but HA isn't and the Bright app is only showing a gas figure for 1am at present (4:30 pm).

**Comment by mcc05** [comment]
Similar issue here too, the integration only seem to work if Bright App/DCC feed is timely. For the past week or so my feed is no where near real-time, it seems that it catches up todays readings in the early hours of the next day, and the integration does not do any catch up and I always get incorrect figures now.

**Comment by gally789** [comment]
Same for me.  Data is not usable now.

**Comment by Prash1407** [comment]
exactly the same issue I am experiencing too. Data is incomplete between the exact time frames as above

**Comment by Markbez** [comment]
My Bright app isn’t much better anymore! This is today!
![D6247DF2-ABB3-4B8F-8458-E3846F50D86E](https://user-images.githubusercontent.com/47301398/219078621-36699830-f297-4a8e-8223-5949a6181301.jpeg)

**Comment by cokecan72** [comment]
Just been directed to this bug from the main discussion. I unfortunately am seeing the exact same issue.

Is there any additional debug logging or similar I can provide from my instance that may help solve it?

Thanks

**Comment by MiralDesai** [comment]
I also contacted Hildebrand and they were talking about a couple of weeks to upgrade their infra, so this issue is likely to be around for a while. They did mention that if you have the CAD then you shouldn't have these issues so I can test that when mine arrives in a few days. It should connect directly with the meter.

**Comment by cokecan72** [comment]
Interesting, so is this an issue at Hildebrand with their infrastructure rather than with this integration?

I am keen to get my hands on one of their SMETS CAD's without display, but been on the waiting list for ages now and still no sign. Get the impression it's in high demand so when their new production run arrives it'll be sold out almost immediately.

**Comment by HandyHat** [comment]
Hey all, as correctly deduced by some of you, this issue lies with the DCC/Bright providing the data later than the 30 minute delay, causing Home Assistant to be further behind and miss out the end of the day.

This seems to have deteriorated as 2023 progresses, and so I have contacted Hildebrand to see if they have any pointers.

I do have an idea on how to resolve this, but I'm not 100% sure it'll work - watch this space!

**Comment by smcfadden** [comment]
I am also experiencing something similar. Difference in readings between Bright and HA over longer periods. See below at images. 
![Screenshot_20230217_185130_Home Assistant](https://user-images.githubusercontent.com/5184751/219865124-1b615635-752d-48ac-9ac0-4f6503284724.jpg)
![Screenshot_20230217_185055_Bright](https://user-images.githubusercontent.com/5184751/219865127-bc5b9430-6db2-43c4-a912-c99a9b570c51.jpg)
PS also notice Bright treats costs as 16bit signed number meaning 32.768 p/kWhr is max and higher than that as with today's exorbitant prices means negative Values. I'll use a template to correct for this 65.536 - (value).

**Comment by daknightuk** [comment]
I had an update this week from Jane at Hildebrand **"This implies your software is making a different api call to get the consumption data than bright does.
If someone tells us exactly what call is not working compared to bright we will have a look.
Ideally please raise a ticket on this specific issue by emailing [support@glowmarkt.com](mailto:support@glowmarkt.com)
The reality is if you want realtime up to date data buy the glowmarkt IHD and get a local mqtt client - everything else is something of a fudge."**

Its a free service so we should not really have high expectations of them but whoever wrote the API probably needs to raise a support ticket and be patient.

**Comment by Markbez** [comment]
That doesn’t explain why their own app isn’t updating until over 24 hours later! Or maybe it does if they are using the same api!

Sent from my iPhone

On 25 Feb 2023, at 06:15, daknightuk ***@***.***> wrote:

﻿

I had an update this week from Jane at Hildebrand "This implies your software is making a different api call to get the consumption data than bright does.
If someone tells us exactly what call is not working compared to bright we will have a look.
Ideally please raise a ticket on this specific issue by emailing ***@***.******@***.***>
The reality is if you want realtime up to date data buy the glowmarkt IHD and get a local mqtt client - everything else is something of a fudge."

Its a free service so we should not really have high expectations of them but whoever wrote the API probably needs to raise a support ticket and be patient.

**Comment by incarvr6** [comment]
So it looks that this integration is useless now... 
Is it only way now to get actual data in real time is using Hildebrand IHD?

**Comment by ethzero** [comment]
> So it looks that this integration is useless now... Is it only way now to get actual data in real time is using Hildebrand IHD?

Honestly, this was my thought too. Not sure if Hildebrand are essentially deprecating the HA/community use of the API.

Possibly related: I was aware people in the past could request for Cloud MQTT access for HA integration, I asked this question (ironically a couple of day before the troubles started) and got this response from Jane Wilson _kinda_ sidestepping the request saying that this could only (now?) be done with their CAD:

> As you realise, **you will need our CAD for MQTT** as it is intended to be used - MQTT is a publish, in near real-time, of data collected on the HAN from your meters by our devices. It is formatted into some json objects and then published for the user to consume.
> 
> There is no (and cannot be an) equivalent of the half hour data for MQTT. If you absolutely need the 30 minute data in MQTT you will need to code against our published API and then inject it into whatever broker you wish.
> 
> API access is automatic; you use your Bright credentials. Further information here : Repository URL - https://bitbucket.org/ijosh/brightglowmarkt/src/master/
> 
> The Bright Application ID is b0f1b774-a586-4f72-9edd-27ead8aa7a8d
> 
> This is our Display CAD: https://shop.glowmarkt.com/products/display-and-cad-combined-for-smart-meter-customers - that comes with local MQTT and cloud MQTT on request. 

All said, this is kinda fair enough honestly I was surprised there was *any* solution to getting detailed (near) real-time access to your own energy data, let alone a free one. While personally I think the price tag of the CAD/IHD is a _little steep_ @ £70 you will end up with a truly local solution and that nicely fit the sprit of Home Assistant, so yeah, I'll probably end up going this route.

**Comment by mcc05** [comment]
would think about the smet2 stick that they have been talking about for a while but never seems to have appeared

**Comment by incarvr6** [comment]
Yeah local connection sounds really good, but £70 is a big price tag. I have Chameleon IHD from Bulb that used to work with ST. Now that integration is totally messed up as well...
I suppose since there is no other options to read our meters I might go for CAD/IHD as well.

**Comment by thazlett** [comment]
I thought it was something I did! I have the same issue :/

May also go via the CAD/IHD route. £70 though..? hmm. What realtime data will that provide? Obviously consumption as you go through the day would be good. But, will it give you realtime instance usage?  Anyone got one that can provide some info here?

**Comment by zbb67** [comment]
Previously I was using a Shelly EM (with a CT clamp on the live feed at the meter).  But after I got a smart meter fitted I switched to this - I thought that the readings would be more accurate if they were read from the meter, and I could also do away with an extra device (the Shelly).

I understand completely that this integration and the API from Hildebrand are all in "best endeavors" and good will, so no complaints from me - but after trying both I've now decided to go back to the Shelly. I was still using it anyway to measure "solar power back into the grid" - a function which Hildebrand say is still in the pipeline as a function for the CAD, if I were to go that route.

The Shelly + clamp were approximately the same cost that the CAD would be, provides near-realtime readings which I think are close enough to the readings from the meter to be good enough for me. It has MQTT and a web interface and integrates into HA.  Minor point, but it is also not tied to the meter so can be taken if I move house.  It does require 240V power (for power and reference) and the clamp might be tight to fit in some meter installations.  It won't suit everyone.

YMMV, but the Shelly works well (enough) for me.  Best of luck to this integration and to Hildebrand though.

Apologies for straying from a bug report, but it might help others here to have another option to consider.

**Comment by Carl-mgb66** [comment]
This is disappointing as it seemed like a good way to get my gas usage - I can get my electricity from my inverter. I'll have to look at other options instead.

**Comment by incarvr6** [comment]
Yeah, this was only way to get gas usage...
I was just about to place an order for CAD IHD and guess what, PayPal payment option is messed up as well...

**Comment by Prash1407** [comment]
I have got a IHD unit at home and it says ZigBee. Anyone can lead me with a tutorial on how to get it into HA?

**Comment by incarvr6** [comment]
Most of IHD operate on zigbee but it is locked for public, so no use for us...

**Comment by daknightuk** [comment]
@Prash1407 I don't think you can use a normal IHD in HA. Zigbee is just the communication protocol in use and from the research I have done its a secure network which you cannot link to.

**Comment by Prash1407** [comment]
> @Prash1407 I don't think you can use a normal IHD in HA. Zigbee is just the communication protocol in use and from the research I have done its a secure network which you cannot link to.

😥 Had a little hope I could get it back to how it used to be, so only way to go this CAD IHD way and then somehow implement?

**Comment by daknightuk** [comment]
So I was going to buy a CAD but after getting solar panels and an inverter there was the realisation that actually the Growatt inveter has all my electricity stats in detail so after a bit of research I came across **Solar Assistant** which is basically a 2nd Pi running a program and using a custom lead to connect to the inverter which took me all of 5 minutes to make. It creates MQTT communication to HA so now I get detailed grid electricity, export to grid, battery stats and solar production and many other stats.
Sadly the CAD is the only solution for me to get my gas stats but to be honest for 4-5 months a year I might just use the E.On IHD and put the meter readings into a database manually for the £70 the CAD costs and of course there is never a guarantee that the technology we buy today will work tomorrow. I've sadly seen the demise of so many cloud providers and services and when the tech becomes a door wedge or paperweight.
My view is that energy providers should use something opensource that people like us can access free of charge without the need for additional hardware or software and just a simple intergration.

**Comment by daknightuk** [comment]
> > @Prash1407 I don't think you can use a normal IHD in HA. Zigbee is just the communication protocol in use and from the research I have done its a secure network which you cannot link to.
> 
> 😥 Had a little hope I could get it back to how it used to be, so only way to go this CAD IHD way and then somehow implement?

Yes the CAD is at the moment the ONLY soution in the marketplace at this moment in time which gives you regular snapshots.

I have seen projects online using optical sensors, magnetic reed switches and even camera technology coupled with OCR technology to get the meter reads but I guess it depends on how far you want to go down the rabbit hole. CAD links to HA through MQTT which to be fair makes it reasonably simple to set up.

**Comment by Prash1407** [comment]
Just realized, mine is actually a IHD-CAD - just went to check it.

![PXL_20230226_232956283](https://user-images.githubusercontent.com/116392836/221444029-bef753e1-cd08-45de-b4a9-8ce0483ff95f.jpg)

**Comment by daknightuk** [comment]
**Consumer access device** that means nothing. 
Hildebrand are the only company offering a CAD that does MQTT communication so without that feature at the moment its not possible to link just any CAD to HA

**Comment by sgreenslade** [comment]
The IHD is the way to go. 
Great youtube video here on how to set it up.
https://youtu.be/r6M9czhUmtA

**Comment by launchm** [comment]
Will the CAD be able to bring in historical data. Thinking of taking the £70 hit

**Comment by gally789** [comment]
> Will the CAD be able to bring in historical data. Thinking of taking the £70 hit

Real time usage only.

**Comment by marcjay** [comment]
> Will the CAD be able to bring in historical data. Thinking of taking the £70 hit

If you don't need the display, and you have SMETS1 meters, the £50 https://shop.glowmarkt.com/products/glow-stick device might be sufficient

**Comment by launchm** [comment]
SMETS2 :( shame there is no USB for that

**Comment by changeofaspect** [comment]
> Will the CAD be able to bring in historical data. Thinking of taking the £70 hit

Based on info sent to me from Glow Support:- 
The CAD provides (almost) realtime power and the meter reading at that point in time, every ten seconds. It does the same for gas and it gives tariff information/standing charge so HA could calculate the cost if you want. For electric, it can differentiate between Import and Export but the Export figure may be limited to the meter reading only - depends on firmware version and meter manufacturer. As well as sending to HA, it also sends this data in realtime to the Bright servers so the Bright App on your phone will also show realtime power.

It does not send any historical data to HA and if it did, I can't see how HA would use that data. Once the required sensors are created and being used, HA will build it's own history going forward but it does not go back in time prior to the date/time when the sensors were created.

The CAD will also calculate the usage for the half-hourly periods we are familiar with in the App and hold this locally in the CAD. Overnight (or some time the next day if the Bright App is anything to go by), the 'official' half -hourly figures held in the DCC are downloaded to the CAD and overwrite the half-hourly figures created by the CAD for the previous day. In this way, the CAD and Bright App will having  matching figures and history.

**Comment by CJDumbleton** [comment]
I recently added this integration to HA and like others I am getting incomplete data. The data I am getting through the Hildebrand app is also incomplete. 

I'm frustrated that it is so difficult / costly to get access to smart meter data. My smart meter data should be freely accessible to me on my phone at all times (not on a stand-alone smart meter display). This discussion has inspired me to start a [petition](https://petition.parliament.uk/petitions/633941/sponsors/new?token=kE7pCqtu3pfGJkvK3_Yt).

**Comment by iaah05** [comment]
> I recently added this integration to HA and like others I am getting incomplete data. The data I am getting through the Hildebrand app is also incomplete.
> 
> I'm frustrated that it is so difficult / costly to get access to smart meter data. My smart meter data should be freely accessible to me on my phone at all times (not on a stand-alone smart meter display). This discussion has inspired me to start a [petition](https://petition.parliament.uk/petitions/633941/sponsors/new?token=kE7pCqtu3pfGJkvK3_Yt).

Signed!

**Comment by changeofaspect** [comment]
> I recently added this integration to HA and like others I am getting incomplete data. The data I am getting through the Hildebrand app is also incomplete.
> 
> I'm frustrated that it is so difficult / costly to get access to smart meter data. My smart meter data should be freely accessible to me on my phone at all times (not on a stand-alone smart meter display). This discussion has inspired me to start a [petition](https://petition.parliament.uk/petitions/633941/sponsors/new?token=kE7pCqtu3pfGJkvK3_Yt).

Signed

**Comment by danq8** [comment]
I bit the bullet and bought the CAD with display (I have a SMETS2 meter) a couple of weeks ago.
It was £73.34 including shipping, it arrived 2 days later. I plugged it in, and was fully operational by the next day.
It is an amazing experience to see near real time data going into Home Assistant. I've set up alerts for when we're feeding back to the grid as our contract means we don't get anything for it, so the investment will pay off for our circumstances. 
Plus, I get the fun of creating "near real time" dashboards showing energy consumed / produced, how our bill is mounting up over the month, and how much we're saving with our solar panels now that we can better manage our usage on the sunny days.
Set up was pretty clear.
I don't think you'll ever get near real time data over the internet, there's just too many points of failure between the 4G SIM cards in the "Smart" meters, the servers at the electric companies moving massive amounts of data, then in this case an extra hop to Hildebrand servers, and then back to your HA. Just sayin'.

**Comment by incarvr6** [comment]
> I recently added this integration to HA and like others I am getting incomplete data. The data I am getting through the Hildebrand app is also incomplete.
> 
> I'm frustrated that it is so difficult / costly to get access to smart meter data. My smart meter data should be freely accessible to me on my phone at all times (not on a stand-alone smart meter display). This discussion has inspired me to start a [petition](https://petition.parliament.uk/petitions/633941/sponsors/new?token=kE7pCqtu3pfGJkvK3_Yt).

Signed

**Comment by incarvr6** [comment]
I seems that integration is intentionally broken for us to buy them IHD...

**Comment by thazlett** [comment]
> I bit the bullet and bought the CAD with display (I have a SMETS2 meter) a couple of weeks ago. It was £73.34 including shipping, it arrived 2 days later. I plugged it in, and was fully operational by the next day. It is an amazing experience to see near real time data going into Home Assistant. I've set up alerts for when we're feeding back to the grid as our contract means we don't get anything for it, so the investment will pay off for our circumstances. Plus, I get the fun of creating "near real time" dashboards showing energy consumed / produced, how our bill is mounting up over the month, and how much we're saving with our solar panels now that we can better manage our usage on the sunny days. Set up was pretty clear. I don't think you'll ever get near real time data over the internet, there's just too many points of failure between the 4G SIM cards in the "Smart" meters, the servers at the electric companies moving massive amounts of data, then in this case an extra hop to Hildebrand servers, and then back to your HA. Just sayin'.

This sounds great. 

I also have some solar panels and do not want to export to the grid. My current methods for measuring grid instant usage is an LDR on the pulsing LED on the actual main unit. This works and lines up with what is on my meter display. But, it is not aware of the directional of flow. So if I suddenly start shifting 300w into the grid, it just shows me 300w grid usage in home assistant. Not 0 or -300w. I have set up some automation to act as power dumps to stop exporting, but it's not perfect.  

Does their CAD show instant data in HASS and, does it differentiate between grid import/export? Or even just show relative positive/negative figure? Buying it now if it does.

**Comment by ethzero** [comment]
> I recently added this integration to HA and like others I am getting incomplete data. The data I am getting through the Hildebrand app is also incomplete.
> 
> I'm frustrated that it is so difficult / costly to get access to smart meter data. My smart meter data should be freely accessible to me on my phone at all times (not on a stand-alone smart meter display). This discussion has inspired me to start a [petition](https://petition.parliament.uk/petitions/633941/sponsors/new?token=kE7pCqtu3pfGJkvK3_Yt).

Also signed. Totally agree: It's fundamentally *our* data that us techy-savvy energy conscience people should be able to access far easier. Hopefully with this grass roots adoption in the future our property's energy data feed will be more open and easier to use for the common person.

**Comment by Nutbolt52** [comment]
I came across this update from Bright about their API. it sheds some light onto the issue and perhaps offers a way forward? 

https://forum.glowmarkt.com/index.php?p=/discussion/347/dcc-data-only-changes

**Comment by incarvr6** [comment]
It seems that readings for current day is back now in integration and more or less same as from local IHD.
I did get them IHD, maybe I rushed but local information is always appreciated, less chance to break...

**Comment by smithbill17** [comment]
So if I've understood everything so far, the issue we have been seeing in Home Assistant with this integration, is not in fact an issue with the integration itself? It's down to some changes made by GlowMarkt on their existing API and that they are awaiting an upgrade to their existing capacity to the DCC (weeks) and until that happens, data can only be retrieved on a 2 hourly basis (will that work for this integration?)

As a result of this issue, I started looking at alternative options (bar paying £70 for a GlowMarkt CAD). My every provider in the UK is Octopus Energy - they have developed a Home Assistant integration (see https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy) but currently this is very limited providing only ONE daily electric & gas reading for the previous 24hrs. However, for their own customers, they are developing 'Octopus Home Mini' (see https://octopus.energy/octopus-home-mini-faq/#ohmfaqs) which is a ZigBee device which connects directly with your SmartMeters and provides 'live' meter data. Currently the 'Octopus Home Mini' is not generally available, but customers can apply to go on their waiting list to receive one (I've no idea how long the wait is).

**Comment by StephenMilner** [comment]
> I am seeing this too. The new algorithm is considerably under reporting compared with the bright app.
> I think the issue may be the new algorithm does not “catch up” if there is a missed reading. The original algorithm caught up so was accurate, albeit putting the consumption in the wrong time slot.

I am seeing exactly the same with Bright App

**Comment by BorjaLL** [comment]
Having the same issue, found out the hard way 

this is a graph comparing DCC data to Home Assistant for the 23rd of Feb
<img width="1319" alt="Monosnap energy csv 2023-03-31 14-26-58" src="https://user-images.githubusercontent.com/3930245/229132889-92334ca9-f37a-482e-beee-9b4946cdfa38.png">

**Comment by TikTok90** [comment]
I'm glad I've found this thread - having just connected to the service I was really wondering what was going on - the blue and purple lines are what my Octopus home mini and my solar system are reporting and the  orange line is the Hildebrand integration.  On the plus side they are working on pulling in the export meter information - which was what I was actually after!
![image](https://github.com/HandyHat/ha-hildebrandglow-dcc/assets/138300803/8d59317d-0c1b-4cd3-b6d8-f746c7e9f119)

**Comment by daknightuk** [comment]
I would steer clear of hildebrand as a data source. I had nothing but problems with their service due to "infrastructure changes" when I contacted them they were quite rude and dismissive and quite frankly I don't deal with rude people so I went for the octopus home mini which is beta but mine arrived after 8 weeks and my partners was about the same. Works well with HA and the data is more accurate and refreshed more often.

**Comment by TikTok90** [comment]
> I would steer clear of Hildebrand as a data source. I had nothing but problems with their service due to "infrastructure changes" when I contacted them they were quite rude and dismissive and quite frankly I don't deal with rude people so I went for the octopus home mini which is beta but mine arrived after 8 weeks and my partners was about the same. Works well with HA and the data is more accurate and refreshed more often.

Not planning on relying on the Hildebrand data - as I already have the mini (for import). That said my dealings with Hildebrand have been absolutely fine, very polite and helpful.  I've also picked up that the API service that they are letting us HA users have is basically a passion project and not their day jobs. It certainly doesn't make them any money - indeed they have had to upgrade their infrastructure at their expense to cope with the number of us HA users using it. So I'm prepared to cut them some slack.

But I agree that they shouldn't have been rude or dismissive to you - no excuse for that.
